workflow/controller/taskresult.go

```diff
@@ -31,6 +31,9 @@ func (wfc *WorkflowController) newWorkflowTaskResultInformer() cache.SharedIndex
 		},
 		func(options *metav1.ListOptions) {
 			options.LabelSelector = labelSelector
+			// `ResourceVersion=0` does not honor the `limit` in API calls, which results in making significant List calls
+			// without `limit`. For details, see https://github.com/argoproj/argo-workflows/pull/11343
+			options.ResourceVersion = ""
 		},
```

Comment:
Out of curiosity, where would this get set to 0 in the first place?

Reply:
and then how about `Limit`? where is that being set?

Reply:
It will be set by the framework during pagination along with offset.

Reply:
Can you help clarify how the UI is enabling that to be called? I guess I'm confused because I thought that creating a `ListWatch` as is being called from `NewFilteredWorkflowTaskResultInformer()` simply performs an initial list and starts a watch on a particular resource.

Reply:
Just spoke to Anton. Sorry, I mistakenly thought when you said "pagination" that you were referring to the UI - hence my question about the UI. He clarified that this is some sort of pagination happening when the Controller is doing listing from the K8S API.

Reply:
So, I'm guessing when say that the `Limit` is being set by the framework, you are referring to the kubernetes client/informer framework, right?

Reply:
Yes the informers framework.